src/pharmpy/distributions.py

```diff
@@ -0,0 +1,38 @@
+from __future__ import annotations
+
+from abc import abstractmethod
+from typing import Union
+
+from pharmpy.deps import numpy as np
+
+
+class NumericDistribution:
```

Comment:
Do we still need these? They are very similar to the "symbolic" distribution classes.

Reply:
What I use is the polymorphism of the `sample` method. It makes sense to build intermediate objects since substituting symbols is so costly (I have not solved that problem yet because it is the `.subs` method of `sympy` `Matrix` objects, and also it's better to reuse built `numpy` objects).

Reply:
The names should be improved.

Reply:
I gave a try at a new polymorphic API to create numeric distribution from symbolic ones: `numeric_dist = symbolic_dist.evalf(parameters)` and then `numeric_dist.sample(...)`.